jib-core/src/main/java/com/google/cloud/tools/jib/frontend/JavaDockerContextGenerator.java

```diff
@@ -259,8 +265,18 @@ public void generate(Path targetDirectory) throws IOException {
       Path directoryInContext = targetDirectory.resolve(copyDirective.directoryInContext);
       Files.createDirectory(directoryInContext);
 
-      // Copies dependencies.
-      FileOperations.copy(copyDirective.sourceFiles, directoryInContext);
+      // Copies the source files.
+      for (LayerEntry layerEntry : copyDirective.layerEntries) {
+        Path destination =
+            directoryInContext.resolve(
+                Paths.get(copyDirective.extractionPath).relativize(layerEntry.getExtractionPath()));
```

Comment:
I'm a bit confused. Shouldn't this be `directoryInContext.resolve(copyDirective.extractionPath).resolve(layerEntry.getExtractionPath())`?

Reply:
`destination` is the location to copy the `layerEntry.sourceFile` to, so it should be resolved to the directory relative to the `directoryInContext`. That relative path is the path from the base extraction path for that `copyDirective` (`copyDirective.extractionPath`) to the actual extraction path for that source file (`layerEntry.extractionPath`).

For example, a `copyDirective` has:
- `directoryInContext = 'classes'`
- `extractionPath = '/app/classes'`

Then, with a context `targetDirectory = 'jib-docker-context'`, `destination` for a `layerEntry.extractionPath = '/app/classes/com/test/HelloWorld'` would be `jib-docker-context/classes/com/test/HelloWorld`.